tools/tkn-results/cmd/root.go

```diff
@@ -21,27 +22,56 @@ import (
 )
 
 var (
+	addr      string
+	authtoken string
+	client    pb.ResultsClient
+
 	//go:embed help.txt
 	help string
 
 	RootCmd = &cobra.Command{
 		Use:   "tkn-results",
 		Short: "tkn CLI plugin for Tekton Results API",
 		Long:  help,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			viper.BindPFlags(cmd.PersistentFlags())
+
+			err := InitResultsClient(*cmd)
+
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
 	}
 )
 
 // Execute executes the root command.
 func Execute() error {
-	return RootCmd.Execute()
+	return RootCmd.ExecuteContext(context.Background())
 }
 
-func client(ctx context.Context) (pb.ResultsClient, error) {
+// Prepare client used to query Kubernetes
+func InitResultsClient(cmd cobra.Command) error {
 	f, err := clientutil.NewDefaultFactory()
+
+	if err != nil {
+		return err
+	}
+
+	client, err = f.Client(cmd.Context())
+
 	if err != nil {
-		return nil, err
+		return err
 	}
-	return f.Client(ctx)
+
+	return nil
+}
+
+func init() {
+	RootCmd.PersistentFlags().StringVarP(&addr, "addr", "a", "", "Result API server address")
+	RootCmd.PersistentFlags().StringVarP(&authtoken, "authtoken", "t", "", "authorization bearer token to use for authenticated requests")
```

Comment:
Same goes for those, could be set on a func that returns the root command and not be part of `init`